packages/preferences/src/browser/preference-frontend-contribution.ts

```diff
@@ -24,34 +35,52 @@ export namespace PreferenceCommands {
     };
 }
 
+export const USER_PREFERENCES_WIDGET_ID = 'user_preferences_widget';
+export const WORKSPACE_PREFERENCES_WIDGET_ID = 'workspace_preferences_widget';
+
 @injectable()
-export class PreferenceFrontendContribution implements CommandContribution, MenuContribution {
+export class PreferencesFrontendContribution<T extends Widget> extends AbstractViewContribution<PreferencesTreeWidget> {
 
     @inject(UserStorageService) protected readonly userStorageService: UserStorageService;
     @inject(PreferenceProvider) @named(PreferenceScope.User) protected readonly userPreferenceProvider: UserPreferenceProvider;
     @inject(PreferenceProvider) @named(PreferenceScope.Workspace) protected readonly workspacePreferenceProvider: WorkspacePreferenceProvider;
     @inject(OpenerService) protected readonly openerService: OpenerService;
     @inject(FileSystem) protected readonly filesystem: FileSystem;
 
+    constructor (protected readonly applicationShell: ApplicationShell, protected readonly options: ViewContributionOptions) {
```

Comment:
Constructor is not necessary. The ApplicationShell should be injected as a property: 

`@inject(ApplicationShell) protected shell: ApplicationShell;`

Reply:
done